### PR TITLE
Momentum based gradient optimizer and optimization techniques for gradient descent

### LINKED ISSUE #4527: [Machine Learning] Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent
#### Proposed title of article
Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent

#### Proposed article introduction
Gradient Descent is an iterative optimization algorithm that finds the smallest value for a function. The general idea is to set the parameters to random values and then iterate in small steps in the direction of the `slope`. Gradient descent is widely used in supervised learning to minimize the error function and find the best parameter values.
Adaptive Optimization (A.O.) Algorithms are becoming more popular due to their ability to converge quickly. In contrast to the traditional Gradient Descent algorithm, all of these algorithms use statistics from previous iterations to strengthen the convergence process. We will be discussing it in detail in this article.
#### Key takeaways 
- Momentum-based Optimization
- Extensions  designed for gradient descent algorithm
- How does momentum help in gradient descent
- Optimization techniques
#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by NdoloBonface**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4527

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by NdoloBonface** [comment]
@lalith1403
Here is the new PR
Duplicate of #4732

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+### Momentum-based optimization
+Momentum-based optimization utilizes an Adaptive Optimization Algorithm (AOA) that uses exponentially-weighted averaging gradients from prior rounds to keep convergence from deviating. Unfortunately, most real-world applications such as deep neural networks rely on noisy input to train.
+
+There is a systematic approach to feeding the data to minimize the impact of noise during optimization. For example, using exponentially weighted averages can solve.
+
+Using an iterative process, we can approximate the trends in a P-dimensional noisy dataset by calculating the parameters as we go through the data:
+
+$$
+On\; iteration\; p\\
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:53]: The new line character isn't working. Can you check that? We can move to final review once this is rectified.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+It's better to use Stochastic Gradient Descent because it fixes many problems. It tries to make up for its inability to do calculations very well. However, it makes a lot more noise than Gradient Descent. Using SGD takes a lot longer to get to a simple solution. It is different from Gradient Descent in that SGD does the derivatives for all points simultaneously.
+
+### Adaptive Moment Estimation
+RMSProp and momentum combine to create this. Like momentum, it saves the declining arithmetic mean of the squared gradients and an average of past gradients, which is similar.
+
+On a smooth surface, it's considerably easier for Adam to make a mistake because of its high coefficient of friction.
+
+This strategy's pseudocode appears like this:
+
+$$
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:78]: Equation is going out of page. Check new line character.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Check the new line character and rectify it. Also shorten the name of the directory containing the article. It is too long.

**Comment by NdoloBonface** [comment]
I'll rectify. Thanks

**Comment by NdoloBonface** [comment]
@lalith1403 I have committed the changes as requested. Thanks

**Comment by lalith1403** [comment]
@NdoloBonface There are merge conflicts in this PR. I suggest you work on the edits, submit a new PR. Thank you.

**Comment by NdoloBonface** [comment]
Okay

**Comment by NdoloBonface** [comment]
@lalith1403 I have pinged you via slack.
